src/main/webapp/react-hooks/use-metacard-definitions.tsx

```diff
@@ -0,0 +1,44 @@
+import { useQuery } from '@apollo/react-hooks'
+import gql from 'graphql-tag'
+import { getIn } from 'immutable'
+
+const query = gql`
+  query MetacardTypes {
+    metacardTypes {
+      id
+      type
+      multivalued
+    }
+  }
+`
+
+//Returns metacard types in map format for easy use
+/*
+ * Ex. {
+ *   created: {
+ *     type: 'DATE',
+ *     id: 'created',
+ *     multivalued: true,
+ *   }
+ *   location: {
+ *     type: 'GEOMETRY',
+ *     id: 'location',
+ *     multivalued: true,
+ *   }
+ * }
+ */
+
+export default () => {
+  const { data, ...rest } = useQuery(query)
+  if (getIn(data, ['metacardTypes'], undefined) === undefined) {
```

Comment:
You might be able to use https://immutable-js.github.io/immutable-js/docs/#/has instead

Reply:
Actually, because you used `getIn` below, you probably don't even need this.

Reply:
true :man_facepalming: